server/service/src/rnr_form/insert.rs

```diff
@@ -170,13 +170,19 @@ fn validate(
             // this should never happen, we've already checked it's there
             .ok_or(InsertRnRFormError::PeriodNotInProgramSchedule)?;
 
-        if previous_period != this_period + 1 {
-            return Err(InsertRnRFormError::PeriodNotNextInSequence);
+        // Periods are ordered by end_date desc in get_schedules_with_periods_by_program, so condition operator implies opposite to what error states
+        if this_period > previous_period {
+            return Err(InsertRnRFormError::PeriodMustBeLaterThanLastUsed);
         }
 
         if form.rnr_form_row.status != RnRFormStatus::Finalised {
             return Err(InsertRnRFormError::PreviousRnRFormNotFinalised);
         }
+
+        // If the the previous form was from many periods ago, don't use it in subsequent calculations
+        if this_period as isize - previous_period as isize > 1 {
```

Comment:
😉 https://doc.rust-lang.org/std/intrinsics/fn.saturating_sub.html or https://docs.rs/num/latest/num/trait.CheckedSub.html

```suggestion
        if this_period.saturating_sub(previous_period) > 1 {
```

Reply:
fair